ros_ws/src/autonomous/src/wall_following.cpp

```diff
@@ -92,7 +92,7 @@ void WallFollowing::followWalls(const sensor_msgs::LaserScan::ConstPtr& lidar)
     float correction = this->m_pid_controller.updateAndGetCorrection(error, TIME_BETWEEN_SCANS);
 
     float steeringAngle = atan(correction) * 2 / M_PI;
-    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - std::max(0.0, std::abs(steeringAngle) - 0.15));
+    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - (float)std::max(0.0, (double)std::abs(steeringAngle) - 0.15));
```

Comment:
```suggestion
    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - std::max(0.0f, std::abs(steeringAngle) - 0.15f));
```
Added the `float` literal to the floating number, which should fix the warning.